### PR TITLE
:book: Update the README

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description

Update the README 

## Motivation
- clarifies project usage and supportability
- Improve readme by adding the links

### CODE REVIEW

**Comment by jmrodri** [comment]
/hold
I want to review this more thoroughly

File: README.md [hunk 1]
```diff
@@ -12,10 +12,26 @@ Kubebuilder increases velocity and reduces the complexity managed by
...
+
+Kubebuilder is extensible and can be imported to be used as a library in other projects.
+Specifically, for those which are looking for a way to use Kubebuilder's existing CLI and scaffolding the projects and re-use its plugins, 
+but to also be able to augment the Kubebuilder project structure with other custom project types so that it is possible to support the Kubebuilder 
+workflow with non-Go operators. 
+
+A good example of its usage is the project [Operator-SDK][operator-sdk]
+that uses Kubebuilder as lib to generate the scaffolds and create its own helper plugins 
+to provide additional features and options on top. [Operator-SDK][operator-sdk] also uses Kubebuilder
+to support their workflow with non-Go operators (_e.g. operator-sdk's Ansible and Helm-based language Operators_)
```
Comment (jmrodri) [inline README.md:32]:
```suggestion
[Operator-SDK][operator-sdk] is a good example of a project that uses Kubebuilder as a library. [Operator-SDK][operator-sdk] uses the plugin feature to include non-Go operators _e.g. operator-sdk's Ansible and Helm-based language Operators_.
```

File: README.md [hunk 2]
```diff
@@ -12,10 +12,26 @@ Kubebuilder increases velocity and reduces the complexity managed by
...
+
+To know more about see [how to create your own plugins][your-own-plugins].
```
Comment (jmrodri) [inline README.md:34]:
```suggestion
To learn more see [how to create your own plugins][your-own-plugins]. 
```

File: README.md [hunk 3]
```diff
@@ -105,7 +121,21 @@ Before starting any work, please either comment on an existing issue, or file a
...
-Currently, Kubebuilder officially supports OSX and Linux platforms. So, if you are using a Windows OS you may find issues. Contributions towards supporting Windows are welcome.
+Currently, Kubebuilder officially supports OSX and Linux platforms. 
+So, if you are using a Windows OS you may find issues. Contributions towards 
+supporting Windows are welcome.
+
+### Apple Silicon
+
+The current scaffold done by the CLI (`go/v3`) when we run `kubebuilder init` uses [kubernetes-sigs/kustomize][kustomize] v3 which does not provide
+a valid binary for Apple Silicon (`darwin/arm64`). Therefore, you can use the `go/v4-alpha` plugin
+instead which provides support for this platform:
```
Comment (jmrodri) [inline README.md:132]: You could simplify this and just let them know that M1 is supported using v4.

```suggestion
Apple Silicon (`darwin/arm64`) is supported using the `go/v4-alpha` plugin which provides support for this platform.
```

If you must let them know why it is broken, then link to an issue. But having the full explanation as to why it doesn't work just makes it wordy.

```suggestion
Apple Silicon (`darwin/arm64`) is supported using the `go/v4-alpha` plugin which provides support for this platform. There is an [issue](https://github.com/kubernetes-sigs/kubebuilder/issues/2566) with the current `go/v3` plugin and Apple Silicon.
```

**Review by jmrodri — CHANGES_REQUESTED** [review]
A few edits I think are warranted.

File: README.md (hunk 3 above)
Comment (camilamacedo86) [reply README.md:132]: @jmrodri it is a good point. Changing

**Pull request merged**